### ISSUE TITLE
packageNamesFromFilePaths not compatible with v2 vs v3 for fragments

### ISSUE THREAD

**Issue by fmatosqg** [apollographql/apollo-kotlin#3860]
**Summary**
Fragments end up in different java packages depending on whether using v2 or v3.

#### following documentation
Documentation states that `packageNamesFromFilePaths("$rootPackageName")` will get us the same result, when it gets us very close but not exactly the same. 

Queries and mutations seem to follow the exact same behavior of v2. However, fragments are generated into package `au.com.woolworths.shop.fragment`, when expected is `au.com.woolworths.shop.graphql.fragment`

![image](https://user-images.githubusercontent.com/5360753/153542359-19216d62-547b-460b-bf01-db49b9a5c1cd.png)

##### note
schema is on different module, and even though trying `packageNamesFromFilePaths("au.com.woolworths.shop.graphql")` on schema module seems to make no difference.

#### packageNameGenerator

When using  packageNameGenerator.set(  .... ) it's not possible to determine where fragments will end up. Somehow, the behavior of `packageName` cannot influence the decision of the package of a fragment kotlin class. This can be confirmed by

1- creating a file called `fragments.graphql` and adding on it a) a query `myQuery` and b) a fragment `myFragment`

2- Write the following function

```
class WowPackageNameGenerator(
    basePackageName: String,
    val queriesPackageName: String,
    val fragmentsPackageName: String
) :
    com.apollographql.apollo3.compiler.PackageNameGenerator {

    private val delegate =
        com.apollographql.apollo3.compiler.PackageNameGenerator.Flat(basePackageName)
    override val version: String
        get() = "0.1"

    override fun packageName(filePath: String): String {

        println("filePath $filePath")

        val p = filePath.replace("/", ".")

        return delegate.packageName(filePath)
            .let {
                "$it.$p"
            }
    }
}

```

Expected: both `myQuery` and `myFragment` end up in the same package.
Result: the filePath which fragment is based is the schema. See screenshot
![image](https://user-images.githubusercontent.com/5360753/153541807-32082b95-e9cd-43ee-9414-268575525d0e.png)

#### workaround

to get the same results as apollo v2 I came up with this

```

class WowPackageNameGenerator2(
    basePackageName: String,
    val queriesPackageName: String,
) :
    com.apollographql.apollo3.compiler.PackageNameGenerator {

    private val delegate =
        com.apollographql.apollo3.compiler.PackageNameGenerator.Flat(basePackageName)
    override val version: String
        get() = "0.1"

    override fun packageName(filePath: String): String {

        return delegate.packageName(filePath)
            .let {
                when {
                    filePath.contains("graphqls") -> "$it.graphql"
                    else -> "$it.$queriesPackageName"
                }
            }
    }
}
```

This is still highly undesirable, since we have +100 modules and most of them run queries and mutations. 

See screenshot for comparison of generated classes. V3 is on left

![image](https://user-images.githubusercontent.com/5360753/153542263-29dc8e8a-bde2-4716-bd6d-2a88dd07dfe5.png)

**Comment by martinbonnin** [comment]
Hi 👋 

With version 3.1.0, you can use `useSchemaPackageNameForFragments` to fallback to 2.x:

```apollo
apollo (
  packageNamesFromFilePaths()
  useSchemaPackageNameForFragments.set(true)
}
```

The 2.x layout creates issues in multi modules scenarios and moving forward, I'd recommend using the simpler `packageName.set("com.example")` option instead of `packageNamesFromFilePaths()` and/or `packageNameProvider` which are a lot more involved

**Comment by fmatosqg** [comment]
Solved, good timing for getting 3.1 out :)
